src/s_midi_alsa.c

```diff
@@ -112,57 +126,69 @@ void sys_alsa_do_open_midi(int nmidiin, int *midiinvec,
     return;
 }
 
-#define md_msglen(x) (((x)<0xC0)?2:((x)<0xE0)?1:((x)<0xF0)?2:\
-    ((x)==0xF2)?2:((x)<0xF4)?1:0)
-
 void sys_alsa_putmidimess(int portno, int a, int b, int c)
 {
+    int status;
     int channel;
     snd_seq_event_t ev;
     snd_seq_ev_clear(&ev);
     if (portno >= 0 && portno < alsa_nmidiout)
     {
-        if (a >= 224)   // pitchbend
-        {
-            channel = a-224;
-            snd_seq_ev_set_pitchbend(&ev, channel, (((c<<7)|b)-8192)); /* b and c are already correct but alsa needs to recalculate them */
-        }
-        else if (a >= 208)      // touch
-        {
-            channel = a-208;
-            snd_seq_ev_set_chanpress(&ev,channel,b);
-        }
-        else if (a >= 192)      // program
-        {
-            channel = a-192;
-            snd_seq_ev_set_pgmchange(&ev,channel,b);
-        }
-        else if (a >= 176)      // controller
-        {
-            channel = a-176;
-            snd_seq_ev_set_controller(&ev,channel,b,c);
-        }
-        else if (a >= 160)      // polytouch
-        {
-            channel = a-160;
-            snd_seq_ev_set_keypress(&ev,channel,b,c);
-        }
-        else if (a >= 144)      // note on
-        {
-            channel = a-144;
-            snd_seq_ev_set_noteon(&ev,channel,b,c);
-        }
-        else if (a >= 128)      // note off
+        status = a & 0xf0;
```

Comment:
so `status <= 0xf0` (240)...

Reply:
The Status value is the upper nibble and the channel is the lower nibble for those messages which have them. This masks out the channel.

Reply:
AH I see what you're saying. That should be:

    status = a & 0xFF;
    if (status < MIDI_SYSEX) status = status & 0xF0;